### DISCUSSION TITLE
How to do routing based on TCP ports when ports are known at runtime?

### DISCUSSION THREAD

**Discussion by kant777** [Kong/kong#10227] [Help]
I deploy applications upon receiving a http request. I want to have gateway/ingress/LB(whatever it is) to routing based on TCP port. For example, Imagine at time t1 I get a http request and I deploy an application A of namspapce A on port 26000 and all the tcp traffic that on port 26000 from external world should directed to application A. similarly, at time t2 I deploy an application B of namespace B on say port 26001 and all the tcp traffic hitting on port 260001 should go to applicationB. how do I achieve this? any tiny example would be great!

**Comment by chronolaw** [comment]
Perhaps you should enable stream listen (https://docs.konghq.com/gateway/latest/reference/configuration/#stream_listen),
and define a tcp route entity(https://docs.konghq.com/gateway/latest/admin-api/#route-object).
